### PR TITLE
fix(db): set error to null when integration is updated

### PR DESCRIPTION
**Pull request by boxbeam**
Also fixes a bug where a provider would not be synced after being updated

### CODE REVIEW

File: ee/tabby-db/src/integrations.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@ impl DbConn {
         };
 
         let res = query!(
-            "UPDATE integrations SET display_name = ?, access_token = ?, api_base = ?, updated_at = DATETIME('now'), synced = false WHERE id = ? AND kind = ?;",
+            "UPDATE integrations SET display_name = ?, access_token = ?, api_base = ?, updated_at = DATETIME('now'), synced = false, error = NULL WHERE id = ? AND kind = ?;",
```
Comment (wsxiaoys) [inline ee/tabby-db/src/integrations.rs:89]: should error only being set to null if access token is not empty？

File: ee/tabby-webserver/src/service/integration.rs [hunk 2]
```diff
@@ -71,6 +71,9 @@ impl IntegrationService for IntegrationServiceImpl {
                 api_base,
             )
             .await?;
+        let _ = self
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/integration.rs:74]: only trigger if access token is not empty?

File: ee/tabby-db/src/integrations.rs (hunk 1 above)
Comment (liangfung) [reply ee/tabby-db/src/integrations.rs:89]: need to consider whether the `api_base` has been modified as well

File: ee/tabby-webserver/src/service/integration.rs [hunk 3]
```diff
@@ -62,6 +62,12 @@ impl IntegrationService for IntegrationServiceImpl {
         access_token: Option<String>,
         api_base: Option<String>,
     ) -> Result<()> {
+        let integration = self.get_integration(id.clone()).await?;
+        let access_token_is_changed = access_token
+            .as_ref()
+            .is_some_and(|token| token == &integration.access_token);
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/integration.rs:68]: should this be `!=`? Please add unit test coverage.

**Pull request merged**